lib/elixir/lib/map.ex

```diff
@@ -27,8 +30,48 @@ defmodule Map do
   @doc """
   Returns a new empty map.
   """
+  @spec new :: map
   def new, do: %{}
 
+  @doc """
+  Creates a `Map` from an enumerable.
+
+  Duplicated keys are removed, the latest one prevails.
+
+  ## Examples
+
+      iex> Map.new([{:b, 1}, {:a, 2}])
+      %{a: 2, b: 1}
+
+      iex> Map.new([a: 1, a: 2, a: 3])
+      %{a: 3}
+
+  """
+  @spec new(Enum.t) :: map
+  def new(enumerable) do
+    Enum.reduce(enumerable, %{}, fn {k, v}, acc -> put(acc, k, v) end)
+  end
+
+  @doc """
+  Creates a `Map` from an enumerable via the transformation function.
+
+  Duplicated entries are removed, the latest one prevails.
+
+  ## Examples
+
+      iex> Map.new([:a, :b], fn (x) -> {x, x} end)
```

Comment:
Folks. This is @vadim-moz's first contribution to Elixir. Although this is important, you have been working on Elixir for years, do we really want to put all of the details we gathered throughout the years into someone's first contribution?

We need to balance this out and I believe @vadim-moz is not supposed to be worried about the usage of comma (or not) or the syntax of anonymous function he clearly got from somewhere else in the codebase.

The code seems correct. The tests cover everything we care about and the documentation is in point. Maybe things could be improved but it is certainly our job, not @vadim-moz. :)